kt/godot-runtime/src/main/kotlin/godot/core/Variant.kt

```diff
@@ -124,17 +124,33 @@ enum class VariantType(
     ),
     STRING(
             { buffer: ByteBuffer, _: Int ->
-                val stringSize = buffer.int
-                val charArray = ByteArray(stringSize)
-                buffer.get(charArray, 0, stringSize)
-                String(charArray, Charsets.UTF_8)
+                val isLong = buffer.bool
+                if(isLong){
+                    val str = LongStringQueue.pollString()
+                    str
+                }
+                else{
+                    val stringSize = buffer.int
+                    val charArray = ByteArray(stringSize)
+                    buffer.get(charArray, 0, stringSize)
+                    val str = String(charArray, Charsets.UTF_8)
+                    str
+                }
             },
             { buffer: ByteBuffer, any: Any ->
                 buffer.variantType = STRING.ordinal
-                any as String
+                require(any is String)
```

Comment:
I don't like this additional check in buffer context. Just cast it, if it is not right type, it will crash later.

Reply:
But it will crash anyway because we will have a TypeCastException

Reply:
Yep, so no need for require

Reply:
Finally we put require everywhere because it's actually more efficient after benchmarking the 2 solutions